### PR TITLE
Deprecate num_proc parameter in  DownloadManager.extract

### LINKED ISSUE #5132: Depracate `num_proc` parameter in `DownloadManager.extract`
The `num_proc` parameter is only present in `DownloadManager.extract` but not in `StreamingDownloadManager.extract`, making it impossible to support streaming in the dataset scripts that use it (`openwebtext` and `the_pile_stack_exchange`). We can avoid this situation by deprecating this parameter and passing `DownloadConfig`'s `num_proc` to `map_nested` instead, as it's done in `DownloadManager.download`.

### PR DESCRIPTION
**Pull request by ayushthe1**
fixes #5132 : Deprecated the `num_proc` parameter in `DownloadManager.extract` by passing `num_proc` parameter to `map_nested` .

### CODE REVIEW

File: src/datasets/download/download_manager.py [hunk 1]
```diff
@@ -377,7 +377,7 @@ def iter_files(self, paths: Union[str, List[str]]):
         """
         return FilesIterable.from_paths(paths)
 
-    def extract(self, path_or_paths, num_proc=None):
+    def extract(self, path_or_paths):
```
Comment (mariosasko) [inline src/datasets/download/download_manager.py:380]: "Deprecating a param" does not mean we can remove the parameter abruptly. This is what we should do instead:
```python
def extract(self, path_or_paths, num_proc="deprecated"):
    """Extract given path(s).

    Args:
        path_or_paths: path or `list`/`dict` of path of file to extract. Each
            path is a `str`.
        num_proc: Use multi-processing if `num_proc` > 1 and the length of
            `path_or_paths` is larger than `num_proc`

            <Deprecated version="2.6.2">

            Pass `DownloadConfig(num_proc=<num_proc>)` to the initializer instead.

            </Deprecated>

    Returns:
        extracted_path(s): `str`, The extracted paths matching the given input
            path_or_paths.

    Example:

    ```py
    >>> downloaded_files = dl_manager.download('https://storage.googleapis.com/seldon-datasets/sentence_polarity_v1/rt-polaritydata.tar.gz')
    >>> extracted_files = dl_manager.extract(downloaded_files)
    ```
    """
    if num_proc != "deprecated":
        warnings.warn(
            "'num_proc' was deprecated in version 2.6.2 and will be removed in 3.0.0. Pass `DownloadConfig(num_proc=<num_proc>)` to the initializer instead.",
            FutureWarning,
        )
    # the rest of the code ...
```

**Review by mariosasko — COMMENTED** [review]
Thanks!

File: src/datasets/download/download_manager.py (hunk 1 above)
Comment (ayushthe1) [reply src/datasets/download/download_manager.py]: Thanks for your help. I commited the changes as you said.

**Comment by ayushthe1** [comment]
Hey @mariosasko . Can you please help me with why the tests keep failing.  I have reviewed the code changes multiple times but can't spot any mistakes.

**Comment by mariosasko** [comment]
You can fix this failure by formatting your code with the `make style` command (run it from the root of the cloned repo).

**Comment by ayushthe1** [comment]
hey @mariosasko ,i cant understand how to use the `make style` command .I searched for it on the internet but cant find any results. 
So i formatted the code using vs-code document formatter. Hope this helps.

**Review by mariosasko — APPROVED** [review]
LGTM!

**Comment by ayushthe1** [comment]
> `make style` runs the "style" target defined here:
> 
> https://github.com/huggingface/datasets/blob/f09f781be3278156ce3aa6ec90c1926b1846a78f/Makefile#L12
> 
> This seems to be a good tutorial on Makefiles: https://opensource.com/article/18/8/what-how-makefile

Thanks! I will look into this :relaxed:

**Pull request merged**